### DISCUSSION TITLE
do something as soon as soon as item get's class"swiper-slide-active"

### DISCUSSION THREAD

**Discussion by Fanni-Portier** [nolimits4web/swiper#4627] [Q&A]
I have short animations (mp4 files) inside of some of the swiper items.
I want them to start playing as soon as the item gets active.
No matter if the item stays in focus or the user scrolls away very quick - the animation should be triggerd.

Currently I use the "activeIndexChange"event.
On this event I detect all ".swiper-slide" items with class ".swiper-slide-active" and find the video element inside.
Then I trigger play with jquery.

My problem is, the animation just starts after the item loses the active class - because the event occurs in the moment the index changes - but the active class is not set in this moment.

So I added a very short timeout - but I guess and hope there is a better way to do this.
Can someone help?
I am working on it only local and with real client data - so I am sorry but I can not share my code here.

**Comment by nolimits4web** [comment]
Maybe `slideChangeTransitionEnd` event fits better for you?

**Reply by Fanni-Portier** [reply]
Thanks, but unfortunately not.
`slideChangeTransitionEnd` occures at the end of the transition.
And I want to see every active item change - no matter how fast you swipe trough.
